nomt/src/beatree/ops/mod.rs

```diff
@@ -84,3 +84,82 @@ fn search_branch(branch: &branch::BranchNode, key: Key) -> Option<(usize, PageNu
     let node_pointer = branch.node_pointer(high - 1);
     Some((high - 1, node_pointer.into()))
 }
+
+#[cfg(feature = "benchmarks")]
+pub mod benches {
+    use crate::{
+        beatree::{
+            branch::{
+                body_size,
+                node::{BranchNode, BranchNodeBuilder},
+                BranchNodePool,
+            },
+            Key,
+        },
+        io::PAGE_SIZE,
+    };
+    use criterion::{BenchmarkId, Criterion};
+    use rand::{Rng, RngCore};
+
+    // Get a vector containing `n` random keys that share the first `shared_bytes`
+    fn get_keys(shared_bytes: usize, n: usize) -> Vec<Key> {
+        let mut rand = rand::thread_rng();
+        let mut prefix = [0; 32];
+        rand.fill_bytes(&mut prefix[0..shared_bytes]);
+
+        let mut keys = vec![];
+        for _ in 0..n {
+            let mut key = prefix.clone();
+            rand.fill_bytes(&mut key[shared_bytes..]);
+            keys.push(key);
+        }
+
+        keys
+    }
+
+    pub fn search_branch_benchmark(c: &mut Criterion) {
+        let mut group = c.benchmark_group("search_branch");
+        let mut rand = rand::thread_rng();
+
+        let branch_node_pool = BranchNodePool::new();
+
+        for prefix_len_bytes in [1, 4, 8, 12, 16] {
+            // TODO: find inverse formula instead of doing it iteratively
```

Comment:
I'd rather not have a TODO in the code unless there's an intention to come after it later. 'nice-to-haves' are better as issues.

Reply:
Yes, I totally agree! We also already discussed how to handle TODOs in the past
I was so focused on finishing the benchmarks that I glossed over the TODO